### PR TITLE
Add description for DM data structures

### PR DESCRIPTION
**Pull request by dr-venkman**
Add description in the DM LWM2M and connectivity header files, to describe data structures.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/include/dm/dm_connectivity.h [hunk 1]
```diff
@@ -35,7 +35,11 @@
 extern "C"
 {
 #endif
-
+/**
+ * @brief Specify the data structure to hold WiFi scanning information.
+ * The information comprises SSID and BSSID of nearby Access Points (APs),
```
Comment (sunghan-chang) [inline framework/include/dm/dm_connectivity.h]: In brief, you should add simple sentence. If you have more description, you should use details.

File: framework/include/dm/dm_lwm2m.h [hunk 2]
```diff
@@ -55,7 +55,9 @@ typedef enum {
 } dm_lwm2m_client_state_e;
 
 /**
- * @brief Struct definition of DM context for a LWM2M session
+ * @brief Specify DM context structure for a LWM2M session.
+ * It comprises LWM2M server information such as ipaddress and port.
+ * Also, it comprises client information such as LWM2M session lifetime.
  */
 
 struct dm_lwm2m_context_s {
```
Comment (sunghan-chang) [inline framework/include/dm/dm_lwm2m.h]: Please split internal structures from here.

File: framework/include/dm/dm_connectivity.h (hunk 1 above)
Comment (jsdosa) [reply framework/include/dm/dm_connectivity.h]: for example)
@brief Specify the data ~
@details The information~~

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**